### ISSUE TITLE
Theme Suggestion: Default Light Theme should be updated to new studio color scheme

### ISSUE THREAD

**Issue by leahcimto** [theforumhelpers/theforumhelpers.github.io#310]
### Theme Title
It is currently called light.

### Colours

**Needs Discussion**
--primary:

--secondary: #7c84ba

--secondaryHover: #485189

--textPrimary: #e6e7ec

--textSecondary:

--linkHover:

--searchBar:

--saturation:

--tertiary:

--tertiaryHover:

--textTertiary:

### Additional context

This theme is inspired by the new thumbnail.

**Comment by penguinmoose** [comment]
Maybe linkHover should be #4a5185 ('Helpers' background?)

**Comment by Accio1** [comment]
I agree that it should be updated to the new style, but the old style should be kept as a theme (probably called "old") so that people can use it if they want.

**Comment by penguinmoose** [comment]
> I agree that it should be updated to the new style, but the old style should be kept as a theme (probably called "old") so that people can use it if they want.

Agree. We don't have to (and shouldn't) delete the old one, so we'll just make a new theme.

**Comment by leahcimto** [comment]
> I agree that it should be updated to the new style, but the old style should be kept as a theme (probably called "old") so that people can use it if they want.

that would be cool.

**Comment by Accio1** [comment]
Using some fancy JS/CSS, it might also be possible to switch the header icon based on theme, so the old theme can have the old icon. Or would we prefer that the header icon just remains the same regardless of theme?

**Comment by leahcimto** [comment]
> Using some fancy JS/CSS, it might also be possible to switch the header icon based on theme, so the old theme can have the old icon. Or would we prefer that the header icon just remains the same regardless of theme?

Changing the icon would be cool too when switched to old mode, nostalgia you know?

**Comment by penguinmoose** [comment]
> Using some fancy JS/CSS, it might also be possible to switch the header icon based on theme, so the old theme can have the old icon. Or would we prefer that the header icon just remains the same regardless of theme?

The easiest way would be to use JS (based on my experience). Whenever the user selects either theme, use JS to change the src attribute of the icon (at least I think that works).

**Comment by penguinmoose** [comment]
> Using some fancy JS/CSS, it might also be possible to switch the header icon based on theme, so the old theme can have the old icon. Or would we prefer that the header icon just remains the same regardless of theme?

It wouldn't be super important to make the icon update based on the theme, though it would look cool if it does.

**Comment by penguinmoose** [comment]
Having the old studio theme reminds me of "old timey mode" in april fools 2022

**Comment by Accio1** [comment]
This is how it looks with the two colours from the thumbnail speech bubbles as primary and secondary colours. I'm not sure if I like it.
![image](https://user-images.githubusercontent.com/48496940/179435611-79f8a91c-71de-49a2-ab99-d31c83ea30d9.png)

**Comment by penguinmoose** [comment]
100% agree. It doesn't feel like the thumbnail at all. Maybe primary would be F8CEF7 and secondary would be EE9D56, using the orangish colors?

**Comment by penguinmoose** [comment]
Wait, what if we make the background a light(er) version of the background of the thumbnail? In other words, I think it would look nice if we make the background the thumbnail without the text with much lighter colors.

**Comment by leahcimto** [comment]
> Wait, what if we make the background a light(er) version of the background of the thumbnail? In other words, I think it would look nice if we make the background the thumbnail without the text with much lighter colors.

I was thinking that could be possible too, maybe even blurred? I don't know how the text would look on that though.

**Comment by penguinmoose** [comment]
> > Wait, what if we make the background a light(er) version of the background of the thumbnail? In other words, I think it would look nice if we make the background the thumbnail without the text with much lighter colors.
> 
> I was thinking that could be possible too, maybe even blurred? I don't know how the text would look on that though.

I thought I said we would remove the text should we set the background to a really light version of the thumbnail.

Blurring might look nice, though I would not say for sure that it would be benificial and look good, because without the text, it is simply a geometric background.

Also, I think we need to make the background thumbnail thing lighter than the actual thumbnail because using the thumbnail directly as the background will probably make it look too sharp (contrast problems and just kinda too strong).

**Comment by Accio1** [comment]
A desaturated version of the thumbnail as the background could be cool. Does anyone want to create a mockup? My concern would be how to do this on longer pages such as the list of Forum Helpers.

**Comment by penguinmoose** [comment]
Something like this?

![TFH](https://user-images.githubusercontent.com/67451414/179608966-3b80d526-137c-46cf-93ac-c67421a0ccfb.png)

I'm thinking we can just make the background not move while scrolling. It should work fine even on really long pages.

**Comment by leahcimto** [comment]
> @Accio1 Something like this?
> 
> ![TFH](https://user-images.githubusercontent.com/67451414/179608966-3b80d526-137c-46cf-93ac-c67421a0ccfb.png)
> 
> I'm thinking we can just make the background not move while scrolling. It should work fine even on really long pages.

I think that's a good idea and it looks cool. We would need to remember to credit the thumbnail artist on the credits page as well.

**Comment by penguinmoose** [comment]
@Accio1 @leahcimto

## Mockups

We need to decide how to fit the image. We really don't want the image to tile. Tiling looks very odd and unprofessional. Not what we want.

### If we scale

We don't want the background to tile, so we need to choose some other way to fit it. If we make it scale (`background-size: 100% 100%`):

<img width="1000" alt="Screen Shot 2022-07-18 at 1 25 19 PM" src="https://user-images.githubusercontent.com/67451414/179611549-198a40a5-aa4f-4f4d-8296-971f9d6c0a93.png">

However, it also means that for devices with other aspect ratios, the background will stretch and squeeze to fit. Here's how it will look on mobile:

<img width="350" alt="Screen Shot 2022-07-18 at 1 30 57 PM" src="https://user-images.githubusercontent.com/67451414/179612067-0b1f2647-0ea0-4d9c-a24e-99e8324fa29c.png">

Looks kinda odd. Since we want the site to work and look good on all devices, this might not be the best option.

### If we make the background big enough to cover

Another option is to make the background big enough to cover (`background-size: cover`), it looks similar to scaling on a computer, but the key difference is on a window or device with an aspect ratio other than the image (like mobile):

<img width="350" alt="Screen Shot 2022-07-18 at 1 38 37 PM" src="https://user-images.githubusercontent.com/67451414/179613272-9839faa8-e1c5-470f-a657-8c42659492bc.png">

The background gets seriously cut off. Not good either.

**What do you think?**

**Comment by penguinmoose** [comment]
Also, we should make it even lighter, judging from my mockups.

**Comment by penguinmoose** [comment]
@Accio1 @leahcimto Can you please look at it and reply? https://github.com/theforumhelpers/theforumhelpers.github.io/issues/310#issuecomment-1188289258

**Comment by leahcimto** [comment]
I've seen your post, I just don't really have anything to say that would add on to the discussion because of the points you already stated.

**Comment by penguinmoose** [comment]
> I've seen your post, I just don't really have anything to say that would add on to the discussion because of the points you already stated. Could I see how cover would look on a computer though?

(sorry if I sounded too harsh; I sometimes get too urgent)

Here you go:

<img width="1000" alt="Screen Shot 2022-07-20 at 11 07 39 AM" src="https://user-images.githubusercontent.com/67451414/180052306-5b60f2c7-a7af-4a8a-9e4b-4903a4c9dfda.png">

It looks pretty much the same as scaling because the aspect ratio of a computer is about the same as the thumbnail's (at least on my computer).

**Comment by penguinmoose** [comment]
Oh, and the other option that can tile is `background-size: contain`. This makes it so that the thumbnail enlarges (but does not stretch or squeeze) as big as possible to still fit within the window, then it tiles that to fill up the missing space.

Computer:

<img width="500" alt="Computer with background-size: contain" src="https://user-images.githubusercontent.com/67451414/180053391-6edc6e35-f779-4815-bed5-0df0b1d89ce9.png">

Mobile:

![Mobile with background-size: contain](https://user-images.githubusercontent.com/67451414/180053598-bd7afbe6-ce0b-4e1a-b0c1-be890f2f3e1e.png)

**Comment by penguinmoose** [comment]
@leahcimto I'd say `cover` is the best option. Also here's a more desaturated version:

<img width="150" src="https://user-images.githubusercontent.com/67451414/180054485-ae4fbbce-1dc4-49f6-b0e9-97b9aa6d016a.png">

**Comment by leahcimto** [comment]
I agree that with our options `cover` sounds like the best one.

**Comment by penguinmoose** [comment]
> I agree that with our options `cover` sounds like the best one.

I mean, having only 2 colors on mobile is still ok (definitely better than having it all squeezed up and deformed).

**Comment by penguinmoose** [comment]
Ok, then I guess `background-size: cover` it is, right?

**Comment by penguinmoose** [comment]
@Accio1 @leahcimto Can someone get this implemented?

**Comment by penguinmoose** [comment]
@Accio1 @leahcimto Can we get going?

The background would be something like this:
```css
background: url("the url to the image") /* we'll host the image on our site. this is just a placeholder */
background-size: cover
```

[This is the link to the background.](https://user-images.githubusercontent.com/67451414/180054485-ae4fbbce-1dc4-49f6-b0e9-97b9aa6d016a.png)

I'm thinking that Secondary could be #2b315c. The rest we can leave as is until we test it.

Can someone at least make a PR? Since I don't know how our CSS variables work, I can't make it (even though I want to).

**Comment by leahcimto** [comment]
Coming up with a complete mockup now.

**Comment by leahcimto** [comment]
Hm, I'm trying to use the image as the background but I'm having some trouble doing that with the CSS variables.

**Comment by penguinmoose** [comment]
@leahcimto hmm.

If `—primary` controls more than just the background, we probably need to make a new CSS variable that controls just the background. Maybe something like `—background`.

For the `background-size:`, hopefully we can just set the background size to `cover` in `body` hardcoded. After all, all the other themes use solid colors so `background-size` should not affect the apperence, except for when there’s an image. So we (probably) won’t need a new variable controlling the background size attribute.

**Comment by penguinmoose** [comment]
@leahcimto Also, does `—secondary: #2b315c` look good? I tested only the header with that color (I don’t know how to do anything with css variables).

**Comment by BarelySmooth** [comment]
This is only my opinion, but having the thumbnail as the website background makes it look a little odd. If this background is only going to be added to the "light theme", it would not remain consistent with the other themes (which do not have a background image). 

Also, the blue secondary color `rgb(49, 172, 186)` (used on the header and other elements) looks out of place with the thumbnail set as the website backdrop. As penguinmoose suggested above, if we instead use `rgb(43, 49, 92)` as the secondary color, the colors don't easily match:
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/83728060/182349223-74877cb7-7054-4cfe-978a-fe6af34d71e8.png">

**Comment by leahcimto** [comment]
I see what you are saying. Maybe the thumbnail background isn't our best option.

**Comment by penguinmoose** [comment]
@BarelySmooth Right. Ok, let's go back a step. Maybe we should just do regular colors.

Or we can maybe make `--secondary` a semitransparent version of that color. That color can be made lighter. Since I still don't know how to use those variables, I can't test, but it's just an idea.

**Comment by penguinmoose** [comment]
@BarelySmooth I see what you mean. That color as secondary doesn't go well with the lightened thumbnail background as background. It looks like it fits more of a dark theme.

Instead, what about a semitransparent version of `rgb(70, 80, 158)`?

**Comment by penguinmoose** [comment]
@leahcimto @BarelySmooth What do you think ^^^?
